unison-core/src/Unison/Referent.hs

```diff
@@ -19,8 +19,15 @@ import qualified Unison.ConstructorType as CT
 
 -- Slightly odd naming. This is the "referent of term name in the codebase",
 -- rather than the target of a Reference.
-data Referent = Ref Reference | Con Reference Int ConstructorType
-  deriving (Show, Ord, Eq)
+type Referent = Referent' Reference
+pattern Ref :: Reference -> Referent
+pattern Ref r = Ref' r
+pattern Con :: Reference -> Int -> ConstructorType -> Referent
+pattern Con r i t = Con' r i t
+{-# COMPLETE Ref, Con #-}
```

Comment:
Did you actually manage to get `COMPLETE` pragma to work? I feel like we tried to use it elsewhere without success.

Reply:
In this case, a clean build got it work for me.  I don't remember which version of ghc we were on when we couldn't get it to work elsewhere compared to now; I could also believe "sometimes it doesn't work".

The issue I remember running into with it in the past was: adding one `COMPLETE` pragma broke another seemingly unrelated one.  (???)  Was it a ghc issue?  a stack issue?

Reply:
@pchiusano I could also have let the actual constructors be called `Ref` and `Con` like they were before instead of `Ref'` and `Con`', but I chose add these patterns so that I could add type annotations to improve whatever inference situations might arise in the future.